File: core/src/main/scala/edu/tufts/hrilab/slug/refResolution/RelevanceTheoreticCandidate.scala
/*
 * Copyright © Thinking Robots, Inc., Tufts University, and others 2024.
 */

package edu.tufts.hrilab.slug.refResolution

import edu.tufts.hrilab.fol.{Variable,Symbol}

import scala.collection.mutable.ListBuffer

/**
* A resolution candidate
 *
* @param ref a reference
* @param relevance the relevance of that candidate
*/
case class RelevanceTheoreticCandidate(ref:Symbol, relevance:Double)

/**
  * Straightforward: A resolution candidate paired with a probability value.
  *
  * @param candidate
  * @param probability
  */
case class RelevanceTheoreticCandidateWithProabability(candidate:RelevanceTheoreticCandidate, probability:Double)

/**
  * A candidate binding between a variable and a candidate
  *
  * @param variable the name of the variable
  * @param candidate the candidate to be bound to that variable
  */
case class RelevanceTheoreticBinding(variable: Variable, candidate: RelevanceTheoreticCandidate)

/**
  * A list of candidates to associate with a given variable
  *
  * @param variable The name of the variable in question
  * @param tiers The list of actions to mnemonic actions to take in searching for the referent of that variable
  * @param candidates The current list of candidates under consideration to be bound to that variable,
  * paired with their associated probability values
  */
case class SingleVarCandidateList(variable: Variable, tiers: ListBuffer[String], candidates: ListBuffer[RelevanceTheoreticCandidateWithProabability])

/**
  * The result of combining together multiple single-variable candidate lists: a set of candidate variable-reference bindings,
  * and an aggregate probability for that joint mapping.
  *
  * @param bindings
  * @param probability
  */
case class CrossMappingCandidateList(bindings:List[RelevanceTheoreticBinding], probability:Double)

File: core/src/main/java/edu/tufts/hrilab/slug/refResolution/RelevanceTheoreticCandidate.java
/*
 * Copyright © Thinking Robots, Inc., Tufts University, and others 2024.
 */

package edu.tufts.hrilab.slug.refResolution;

import edu.tufts.hrilab.fol.Symbol;

/**
 * A resolution candidate
 */
public class RelevanceTheoreticCandidate {
  /**
   * a reference
   */
  private final Symbol ref;
  /**
   * the relevance of that candidate
   */
  private final double relevance;

  public RelevanceTheoreticCandidate(Symbol ref, double relevance) {
    this.ref = ref;
    this.relevance = relevance;
  }

  public Symbol ref() {
    return ref;
  }

  public double relevance() {
    return relevance;
  }

  @Override
  public String toString() {
    return "RelevanceTheoreticCandidate(" + ref + ", " + relevance + ")";
  }
}
